Add documentation for max_number_processors (#8157)
opensearch-project/documentation-website: _ingest-pipelines/processors/index-processors.md

`urldecode` | Decodes a string from URL-encoded format.
`user_agent` | Extracts details from the user agent sent by a browser to its web requests. 

> ## Validations on Processors

Reviewer: Can we call it as "Processor limit settings"?

Author: Updated.

> Configure limits on the number of ingest processors using the cluster setting `cluster.ingest.max_number_processors`. The total number of processors includes both the number of processors and the number of [`on_failure`]({{site.url}}{{site.baseurl}}/ingest-pipelines/pipeline-failures/) processors.

Reviewer 2: ```suggestion
You can limit the number of ingest processors using the cluster setting `cluster.ingest.max_number_processors`. The total number of processors includes both the number of processors and the number of [`on_failure`]({{site.url}}{{site.baseurl}}/ingest-pipelines/pipeline-failures/) processors.
```

> We can configure the limits on the number of ingest processors that should be used. The limit can be configured using the attribute `cluster.ingest.max_number_processors`. The sum of the number of processors and the number of `on_failure` processors are considered for counting the total number of processors on which the limit would be applied.

Reviewer: > The limit can be configured using the attribute

The limit can be configured using the cluster setting

Author: Updated

> The default value for `cluster.ingest.max_number_processors` is `Integer.MAX_VALUE`. Adding more processors than the configured value in `cluster.ingest.max_number_processors` will throw an `IllegalStateException`.

Reviewer 2: ```suggestion
The default value for `cluster.ingest.max_number_processors` is `Integer.MAX_VALUE`. Adding a higher number of processors than the value configured in `cluster.ingest.max_number_processors` will throw an `IllegalStateException`.
```
